remote.go

```diff
@@ -18,6 +18,23 @@ import (
 	"unsafe"
 )
 
+type RemoteCreate uint
+
+const (
+	// Ignore the repository apply.insteadOf configuration
+	RemoteCreateSkipInsteadof RemoteCreate = C.GIT_REMOTE_CREATE_SKIP_INSTEADOF
+	// Don't build a fetchspec from the name if none is set
+	RemoteCreateSkipDefaultFetchspec RemoteCreate = C.GIT_REMOTE_CREATE_SKIP_DEFAULT_FETCHSPEC
+)
+
+// RemoteCreateOptions contains options for creating a remote
+type RemoteCreateOptions struct {
+	Repository *Repository
```

Comment:
```suggestion
```

it's probably better to not have this be passed in through the `Options` object, since the pointer to the C repository can be obtained from within `CreateWithOptions`, which has a strong affinity to a pre-existing `git.Repository` and would be confusing (API-wise) that one would be able to call `myOneRepository.Remotes.CreateWithOptions(...)` and the resulting remote would be associated with a repository that is not `myOneRepository`.

in order to support detached remotes, we could introduce `func NewDetachedRemote(url string, options *RemoteCreateOptions) (*Remote, error)`.

Reply:
In my opinion, `NewDetachedRemote` is not needed right now. That was my design mistake. (Due to a dead copy of the C interface).